### PR TITLE
Add GenesisBlock::OperatingMode to control how cluster features are activated

### LINKED ISSUE #5757: Design/implement mechanism to enable cluster features over time (block based)
We want to be able to unlock cluster features over time for initial mainet launch.   Initially only voting and staking, then system transfers, and more.   The solution should be block height based, and likely can be hard coded (ie, validators must update their builds, aka hard fork) to unlock new features.

### LINKED ISSUE #6370: Potentially insecure native programs available to SLP1
#### Problem

Potentially insecure native programs available to SLP1

#### Proposed Changes

Chop out anything nonessential, such as Budget, which is part of the CLI, but likely unused.

### PR DESCRIPTION
**Pull request by mvines**
`solana-genesis --dev` selects the `Development` operating mode, which enables all programs.  Without the `--dev` flag, `solana-genesis` selects the `SoftLaunch` operating mode.

TODO:
* [x] Add support for enabling new native programs at arbitrary epoches
* [x] Tests

Fixes #5757
Fixes #6370

### CODE REVIEW

**Comment by mvines** [comment]
stable-perf failure is due to the flaky GPU CI machines

File: genesis_programs/src/lib.rs [hunk 1]
```diff
@@ -21,43 +23,107 @@ extern crate solana_vest_program;
...
+                None
+            }
+        }
+        OperatingMode::SoftLaunch => {
+            if epoch == 0 {
+                // Voting and Staking only at epoch 0
+                Some(vec![
+                    solana_config_program!(),
+                    solana_stake_program!(),
+                    solana_vest_program!(),
```
Comment (garious) [inline genesis_programs/src/lib.rs]: Let's kill `vest` and `config` for now. Anatoly thinks we can do without.

Comment (mvines) [reply genesis_programs/src/lib.rs]: Done

File: sdk/src/genesis_block.rs [hunk 2]
```diff
@@ -33,6 +39,7 @@ pub struct GenesisBlock {
     pub rent_calculator: RentCalculator,
     pub inflation: Inflation,
     pub epoch_schedule: EpochSchedule,
+    pub operating_mode: OperatingMode,
```
Comment (rob-solana) [inline sdk/src/genesis_block.rs]: should we expose this to developers (like make it a sysvar)?

Comment (mvines) [reply sdk/src/genesis_block.rs]: Can you think of a use case for why that'd be helpful?  I don't see a strong need right now

Comment (rob-solana) [reply sdk/src/genesis_block.rs]: only https://github.com/solana-labs/solana/issues/6193

Comment (rob-solana) [reply sdk/src/genesis_block.rs]: no use cases, just a general issue of programs not being able to see network config

Comment (mvines) [reply sdk/src/genesis_block.rs]: 🤷‍♂ This is "meta config".  A program that wants to do something based on OperatingMode is probably completely broken

**Pull request merged**